modules/graphql_voyager/assets/src/index.js

```diff
@@ -0,0 +1,47 @@
+import React, { Component } from 'react';
+import ReactDOM from 'react-dom';
+import { Voyager } from 'graphql-voyager';
+import fetch from 'isomorphic-fetch';
+import Drupal from 'drupal';
+import jQuery from 'jquery';
+
+/**
+ * Behavior for rendering the GraphQL Voyager interface.
+ */
+Drupal.behaviors.graphQLRenderVoyager = {
+  attach: (context, settings) => {
+    const container = jQuery('#graphql-voyager', context).once('graphql-voyager')[0] || undefined;
+    const INTROSPECTION_URL = settings.graphQLRequestUrl || `${window.location.origin}/graphql`;
+
+    if (typeof container === 'undefined') {
+      return;
+    }
+
+    class VoyagerWrapper extends Component {
+      constructor() {
+        super();
+      }
+
+      render() {
+        return (
+          <Voyager introspection={this.introspectionProvider} displayOptions={{ skipRelay: true }}/>
+        );
+      }
+
+      introspectionProvider(query) {
```

Comment:
Let's provide the introspection .json through the graphql service directly and then forwarding it with Drupal.settings instead of fetching it here. That way we can make the voyager work without having to rely on the person that visits the page to have access to execute arbitrary graphql queries. We should actually do the same with the graphiql explorer.